### PR TITLE
[LIT-178] Create Version 4.3.0

### PR DESCRIPTION
**Pull request by edaroit**
### Changes

- No more message-contracts (from previous 4.2.x versions)
- Updated `Formatter`
- Updated `Credo`
- `Poison` replaced by `Jason`

### CODE REVIEW

File: lib/kafka_message_bus/producer.ex [hunk 1]
```diff
@@ -34,22 +40,42 @@ defmodule KafkaMessageBus.Producer do
...
+    |> Map.to_list()
+    |> Enum.reject(&is_invalid_element?/1)
+    |> Map.new()
+  end
+
+  defp remove_invalid_elements(data), do: data
+
+  defp is_invalid_element?(element) do
+    case element do
+      {key, _} when key in [:__meta__, :__struct__] -> true
```
Comment (cgerling) [inline lib/kafka_message_bus/producer.ex:63]: We could move this into an `invalid_element_keys` (or something better) module attribute

**Review by cgerling — CHANGES_REQUESTED** [review]
Although it's a small PR we could split it into even smaller ones because there are 3 distinct objectives here: add credo and fix issues, change code formatting (both config and source code), and some changes needed to `4.3.0`.

**Comment by edaroit** [comment]
@cgerling The idea is to have everything on the same PR so we may release the new library version which had issues at the `4.2.x` versions because `message_contracts` and the code itself wasn't stable.

File: lib/kafka_message_bus/producer.ex [hunk 2]
```diff
@@ -34,22 +40,42 @@ defmodule KafkaMessageBus.Producer do
...
   end
 
+  defp remove_invalid_elements(data) when is_map(data) do
+    data
+    |> Map.to_list()
+    |> Enum.reject(&is_invalid_element?/1)
+    |> Map.new()
+  end
+
+  defp remove_invalid_elements(data), do: data
```
[the remark this answers is not in the document]
Comment (edaroit) [reply lib/kafka_message_bus/producer.ex:59]: Why would we want this?

File: lib/kafka_message_bus/producer.ex (hunk 1 above)
Comment (edaroit) [reply lib/kafka_message_bus/producer.ex:63]: Good call, I'll update this.

File: lib/kafka_message_bus/producer.ex (hunk 2 above)
[the remark this answers is not in the document]
Comment (cgerling) [reply lib/kafka_message_bus/producer.ex:61]: My question was kind of ambiguous, let me rephrase it
Do we allow producing messages with data that aren't maps?
Because if we don't, we could remove this case but if we do, could we add test cases to clarify this behavior?

[the remark this answers is not in the document]
Comment (edaroit) [reply lib/kafka_message_bus/producer.ex:61]: Actually, this behavior did not exist before. So I don't think it's necessary. Good catch

**Comment by cgerling** [comment]
Just to clarify, I wasn't aware (bad memory :P) that the changes related to Credo and formatting were already made and was just cherry-picked to this branch because every change after (and including) `4.2.0` was "archived" so it isn't worth to split this into new PRs, thanks for explaining that for me @edaroit

**Review by cgerling — APPROVED** [review]

**Review by cgerling — APPROVED** [review]
